src/data/commandLine.js

```diff
@@ -15,20 +16,34 @@ var find = (program) => GLib.find_program_in_path(program);
 
 /**
  * @param {string} command - the command as String (e.g. "ps aux")
+ * @params {string} [input] - optional input for stdin
  * @return {Promise} the command execution result as a String, or fails if an error occur
  */
 /* exported execute */
-var execute = (command) => new Promise((resolve, reject) => {
+var execute = (command, stdin = null) => new Promise((resolve, reject) => {
     Log.d(LOGTAG, `Executing: "${command}"`);
-    const [_, stdout, stderr, status] = GLib.spawn_command_line_sync(command);
-    if (status !== 0) {
-        const error = stderr.toString();
-        Log.e(LOGTAG, error);
-        reject(error);
-    }
-    const output = stdout.toString();
-    Log.d(LOGTAG, `Output: ${output}`);
-    resolve(output);
+
+    let proc = Gio.Subprocess({
```

Comment:
I guess that here it should probably be: `let proc = new Gio.Subprocess({`
Changed.

Reply:
Oops, sorry. I always forget the new. :P